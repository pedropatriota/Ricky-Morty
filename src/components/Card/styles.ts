import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 18rem;
`;

export const ImagContainer = styled.div<{ image: string }>`
	background-image: url(${props => props.image});
	background-repeat: no-repeat;
	background-size: cover;
	width: 100%;
	max-width: 18rem;
	height: 18rem;
	border-radius: 1rem 1rem 0 0;
`;

export const InfoContainer = styled.div`
	padding: 0.5rem;
	background-color: ${props => props.theme.colors.el};
	border: 1px solid ${props => props.theme.colors.el};
	border-radius: 0 0 1rem 1rem;

	h2 {
		margin-bottom: 1rem;
		color: ${props => props.theme.colors.text};
	}
	p {
		margin-bottom: 0.4rem;
		color: ${props => props.theme.colors.text};
	}
`;
