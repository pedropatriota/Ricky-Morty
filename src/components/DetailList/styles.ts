import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.75rem;
	flex-direction: column;
	margin-bottom: 0.75rem;
`;
