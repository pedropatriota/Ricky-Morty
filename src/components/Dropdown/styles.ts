import styled from 'styled-components';
import Select from 'react-select';

export const ReactSelect = styled(Select)`
  .react-select__control {
    width: min(90vw, 12.5rem);
    background-color: ${(props) => props.theme.colors.el};
    border-color: ${(props) => props.theme.colors.border};
    box-shadow: 0 0 0 transparent;
    height: 3.375rem;

    &:hover {
      border-color: ${(props) => props.theme.colors.border};
    }
  }

  .react-select__single-value {
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__menu {
    width: 12.5rem;
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__option,
  .react-select__option--is-focused 
  {
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__dropdown-indicator,
  .react-select__indicator-separator {
    display: none;
  }
`;
