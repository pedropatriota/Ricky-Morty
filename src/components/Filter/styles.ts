import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	align-items: flex-start;
	justify-content: space-between;
	margin-top: 3.1rem;
	padding: 0 min(5vw, 5rem);
	flex-wrap: nowrap;
	gap: 0.5rem;

	@media (max-width: 500px) {
		flex-wrap: wrap;
		gap: 1rem;
	}
`;
