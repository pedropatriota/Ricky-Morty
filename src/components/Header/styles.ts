import styled from 'styled-components';

export const Header = styled.header`
	padding: 0px min(5vw, 5rem);
	background-color: ${({ theme }) => theme.colors.bg};
	box-shadow: 0px 10px 15px -3px rgba(0, 0, 0, 0.1);
	margin-bottom: 4rem;
`;

export const Container = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	flex-wrap: wrap;

	img {
		width: 100%;
		height: auto;
		max-width: 10rem;
	}
`;
