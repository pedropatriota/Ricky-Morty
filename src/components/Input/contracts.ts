import { InputHTMLAttributes } from 'react';

export interface IFormProps extends InputHTMLAttributes<HTMLInputElement> {
	value?: string;
	handleChange(event: React.ChangeEvent<HTMLInputElement>): void;
}
