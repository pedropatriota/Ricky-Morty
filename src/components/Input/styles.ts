import styled from 'styled-components';

export const Input = styled.input`
  width: min(90vw, 30rem);
  border: ${(props) => props.theme.colors.border};
  color: ${(props) => props.theme.colors.text};
  height: 3.375rem;
  border-radius: 3px;
  padding: 0 2rem;
  font-size:0.94rem;
  outline: none;
  background: ${(props) => props.theme.colors.el}
`;
