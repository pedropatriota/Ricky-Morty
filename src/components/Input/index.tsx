import { InputHTMLAttributes } from 'react';
import { Input as InputEl } from './styles';

export interface IFormProps extends InputHTMLAttributes<HTMLInputElement> {
  value: string;
  handleChange(event: React.ChangeEvent<HTMLInputElement>): void;
}

const Input = ({ value, handleChange }: IFormProps) => {
  return (
    <InputEl type="text" value={value} onChange={handleChange} placeholder="Search for a country" />
  );
};

export default Input;
