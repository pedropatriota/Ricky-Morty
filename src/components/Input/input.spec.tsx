import Input from '.';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';
import { dark } from '../../styles/themes';

const handleChangeMocked = jest.fn();
describe('Input', () => {
  it('should render correctly', () => {
    render(
      <ThemeProvider theme={dark}>
        <Input value="" handleChange={handleChangeMocked} />
      </ThemeProvider>
    );

    expect(screen.getByRole('textbox')).toBeVisible();
    expect(screen.getByRole('textbox')).toHaveValue('');
    expect(screen.getByPlaceholderText('Search for a country')).toBeInTheDocument();
  });

  it('should be able to type in the input', async () => {
    render(
      <ThemeProvider theme={dark}>
        <Input value="" handleChange={handleChangeMocked} />
      </ThemeProvider>
    );

    const inputElement = screen.getByPlaceholderText('Search for a country') as HTMLInputElement;

    await waitFor(async () => {
      await userEvent.type(inputElement, 'World!');
      expect(handleChangeMocked).toHaveBeenCalled();
    });
  });

  it('should match the snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={dark}>
        <Input value="" handleChange={handleChangeMocked} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
