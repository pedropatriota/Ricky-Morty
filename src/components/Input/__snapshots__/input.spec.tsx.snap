// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the snapshot 1`] = `
<div>
  <input
    class="sc-aXZVg iVjMKk"
    placeholder="Search for a country"
    type="text"
    value=""
  />
</div>
`;
